src/output.rs

```diff
@@ -66,10 +74,16 @@ impl Output {
     }
 
     pub fn write_host(&mut self, host: &str) -> anyhow::Result<()> {
+        if self.deduplicate {
+            if let Some(_) = self.seen.get(host) {
```

Comment:
This check can be written more clearly as:
```suggestion
if self.seen.contains(host) {
```